src/main.c

```diff
@@ -2403,87 +2392,53 @@ int main(int argc, char **argv) {
         WSACleanup();
         return 1;
     }
-#endif
 
-    // CHECK COMMAND LINE ARGUMENTS //
-    if (argc < 4 || argc > 5) {
-        printf("USAGE: hostname nick password [port]\n");
-        exit(1);
-    }
-
-    // INITIALIZATION //
-    srand(time(NULL));
-    rand();
+    return 0;
+}
+#else
+int init_winsock() { return 0; }
+#endif
 
+int init_initory() {
     inventory.items = calloc(INVENTORY_SLOTS * INVENTORY_ROWS, sizeof(Item));
     for (int item = 0; item < INVENTORY_SLOTS * INVENTORY_ROWS; item ++) {
         inventory.items[item].id = 0;
         inventory.items[item].num = 0;
     }
     inventory.selected = 0;
+}
 
-    g->blocks_recv = 0;
-    g->debug_screen = 0;
-    g->inventory_screen = 0;
-    g->chunk_buffer_size = CHUNK_SIZE*CHUNK_SIZE*CHUNK_SIZE;
-    g->chunk_buffer = malloc(sizeof(char)*g->chunk_buffer_size);
+char* shtxt_path(const char *name, const char *type, char *path) {
```

Comment:
Best practice would be to also take an size_t size variable that is the maximum size of the path *. Then you should use snprintf rather than sprintf with this limit.